test/unit/lib/plugins/aws/package/compile/events/kafka.test.js

```diff
@@ -149,341 +149,372 @@ describe('test/unit/lib/plugins/aws/package/compile/events/kafka.test.js', () =>
-      await expect(
-        runServerless({
-          fixture: 'function',
-          configExt: {
-            functions: {
-              basic: {
-                events: [
-                  {
-                    kafka: {
-                      topic,
-                      bootstrapServers: ['abc.xyz:9092'],
-                      accessConfigurations: { vpcSubnet: 'subnet-abc4567890' },
+      it('should fail to compile EventSourceMapping resource properties for VPC_SUBNET with no VPC_SECURITY GROUP', async () => {
+        await expect(
+          runServerless({
+            fixture: 'function',
+            configExt: {
+              functions: {
+                noVpcSecurityGroup: {
+                  events: [
+                    {
+                      kafka: {
+                        topic,
+                        bootstrapServers: ['abc.xyz:9092'],
+                        accessConfigurations: { vpcSubnet: 'subnet-abc4567890' },
+                      },
                     },
-                  },
-                ],
+                  ],
+                  handler: 'path.to.handler',
+                },
               },
             },
-          },
-          command: 'package',
-        })
-      ).to.be.rejected.and.eventually.have.property(
-        'code',
-        'INVALID_NON_SCHEMA_COMPLIANT_CONFIGURATION'
-      );
-    });
+            command: 'package',
+          })
+        ).to.be.rejected.and.eventually.contain({
+          code: 'FUNCTION_KAFKA_VPC_ACCESS_CONFIGURATION_MISSING_VPCSECURITYGROUP',
+          message:
+            'You must specify at least one vpcSecurityGroup accessConfiguration for function: noVpcSecurityGroup',
+        });
+      });
 
-    it('should fail to compile EventSourceMapping resource properties for VPC_SECURITY GROUP with no VPC_SUBNET', async () => {
-      await expect(
-        runServerless({
-          fixture: 'function',
-          configExt: {
-            functions: {
-              basic: {
-                events: [
-                  {
-                    kafka: {
-                      topic,
-                      bootstrapServers: ['abc.xyz:9092'],
-                      accessConfigurations: { vpcSecurityGroup: 'sg-abc4567890' },
+      it('should fail to compile EventSourceMapping resource properties for VPC_SECURITY GROUP with no VPC_SUBNET', async () => {
+        await expect(
+          runServerless({
+            fixture: 'function',
+            configExt: {
+              functions: {
+                noVpcSubnet: {
+                  events: [
+                    {
+                      kafka: {
+                        topic,
+                        bootstrapServers: ['abc.xyz:9092'],
+                        accessConfigurations: { vpcSecurityGroup: 'sg-abc4567890' },
+                      },
                     },
-                  },
-                ],
+                  ],
+                  handler: 'path.to.handler',
```

Comment:
@pgrzesik @lewgordon for some reason these tests would only pass when a handler was specified, even though this property is not specified in other very similar tests in this file. I could not figure out why; any insight?

Reply:
This is because in other cases, the configuration was extended for function already defined as a part of the fixture that had `handler` defined, here you added a new function which requires `handler` to be defined manually

Reply:
got it, thanks. renaming to `basic` fixed it 😁